server/tests/tram_track_graph_transformer/test_tram_track_graph_transformer.py

```diff
@@ -0,0 +1,190 @@
+from collections import deque
+from math import sqrt
+
+import overpy
+import pytest
+from networkx import DiGraph
+from pyproj import Geod
+from src.tram_track_graph_transformer import TramTrackGraphTransformer
+from src.tram_track_graph_transformer.node import Node
+
+
+@pytest.fixture()
+def graph_25(osm_tram_stops_and_tracks, krakow_city_configuration):
+    transformer = TramTrackGraphTransformer(
+        osm_tram_stops_and_tracks, krakow_city_configuration
+    )
+    return (transformer, transformer.densify_graph_by_max_distance(25.0))
+
+
+class TestTramTrackGraphTransformer:
+    CORRECT_MAX_DENSIFICATION_DISTANCES = [10.0, 25.0]
+    INCORRECT_MAX_DENSIFICATION_DISTANCES = [-5.0, 0.0]
+    _geod = Geod(ellps="WGS84")
+
+    def test_tram_track_transformer_crossings_in_graph(
+        self,
+        krakow_ignored_crossings_and_stops,
+        osm_tram_track_crossings: overpy.Result,
+        graph_25: tuple[TramTrackGraphTransformer, DiGraph],
+    ):
+        # arrange
+        _, graph = graph_25
+
+        for node_id in osm_tram_track_crossings.get_node_ids():
+            if (
+                node_id
+                not in krakow_ignored_crossings_and_stops["ignored_crossings_ids"]
+            ):
+
+                # assert
+                assert graph.has_node(node_id) is True
```

Comment:
I believe there's no need to be that specific:
```suggestion
                assert graph.has_node(node_id)
```
(Applies to the entire file)